src/listeners/interactionCreate.ts

```diff
@@ -136,8 +137,15 @@ client.on("interactionCreate", async interaction => {
 								console.error(`Couldn't get string ${path} in English for ${file}, please fix this`)
 						}
 					}
-					if (typeof string === "string" && variables)
-						for (const [variable, text] of Object.entries(variables)) string = string.replace(`%%${variable}%%`, String(text))
+					if (typeof string === "string" && variables) {
+						const discordLocale = transformBotLocale(lang) ?? "en-GB"
+						string = new MessageFormat(
+							{
+								[discordLocale]: (value: string | number, ord?: boolean) =>
+									new Intl.PluralRules(discordLocale, { type: ord ? "ordinal" : "cardinal" }).select(Number(value)),
+							}[discordLocale],
```

Comment:
you're creating an object with 1 key just to index it straight after? Why?

Reply:
> you're creating an object with 1 key just to index it straight after? Why?

Default plural function is throws an error if an unsupported form for locale was used (zero, one, few, etc).
To make a custom plural function you need to name it by the locale. I don't know another way to name it with a variable.

Reply:
you can create a function without making it in an object

Reply:
The name of the function should be a locale. How should I take the name of the function from a variable?